avsomr/avsomrdrawer.cpp

```diff
@@ -0,0 +1,208 @@
+#include "avsomrdrawer.h"
+
+#include <QPainter>
+
+#include "libmscore/score.h"
+#include "libmscore/measure.h"
+
+#include "avslog.h"
+
+using namespace Ms::Avs;
+
+AvsOmrDrawer::AvsOmrDrawer()
+      {
+
+      }
+
+//---------------------------------------------------------
+//   makeContext
+//---------------------------------------------------------
+
+std::shared_ptr<AvsOmrDrawer::Context> AvsOmrDrawer::makeContext(QPainter* p, const Ms::Score* score) const
+      {
+      IF_ASSERT(p) {
+            return nullptr;
+            }
+
+      const Ms::MasterScore* ms = score->masterScore();
```

Comment:
Why not use `namespace Ms` in this file, but instead mention `Ms` class names like this?

Reply:
Usually, I add the using namespace to what is part of the current module, and explicitly specify the namespace to external classes relative to the current module. With some exceptions, when an external class is used a lot and in many places.

Reply:
Actually, I don't understand why the namespace logic is not putting all files except the import one into a simple `Avs` namespace, just like `Ove` (currently `OVE`, but this should be changed) for overture file. Why do you need `Ms::Avs`? In my opinion it's simple and logic enough to just use `Avs` for all audiveris related code. Just like the files in `mscore` module, you put `namespace Avs {` at the beginning and `}` at the end.

Reply:
Strictly speaking, this is the logic of working with Audiveris *within* MuseScore, hence there is nothing wrong in adding it to `Ms` namespace. Adding a separate namespace for this exact module is also a good thing to do. Hence there is a `Ms::Avs` namespace like implemented here. A plain `Avs` namespace would also do a job, but why would this option be so much preferable?

Reply:
I wouldn't say "so much preferable", it's just if you put other namespaces under `Ms`, then why do you need `Ms` from the very beginning? To my understanding, `Ms` is for MuseScore score data model and UI codes. Other things such as importing and exporting should be in different namespaces.

Reply:
One of the important reasons to have `Ms` namespace is to separate MuseScore code from third-party libraries code which may use different namespaces or no namespaces at all.

Reply:
Usually, the namespaces are made in the format: company.application.module.submodule  
(this is especially evident in other languages)
for simplicity, sometimes a namespace company is omitted.  
I noticed that Ms namespace is used everywhere, and in libmscore and mscore, so I decided that Ms is the application namespace.
And Avs is a module in the application.
So it turns out Ms::Avs

Reply:
Oh, I didn't think of that. Then I guess it's OK to use `Avs` inside `Ms`... But we've already done other namespaces, like `Awl`, `Ove` (currently `OVE`) outside `Ms`, it's more consistent to put this one outside too.

Reply:
If Ove, Awl are separate, unrelated to MU libraries (like thirdparty), then it’s correct that they have a non-Ms namespace. If they are part of MU, then they should be in the Ms namespace, and it’s better to fix it. (I can do it myself later)

Reply:
`Awl` is audio widget library, it just contains custom Qt widgets that're used in `Ms`, so I guess you can call it "unrelated"? `Ove` is for importing Overture files, I don't know whether it's "related" or not. But `Ove` is just like `Avs`, by any means they should have the same level.

Reply:
As far as I understand Awl historically came from another Werner's project so it might be considered somewhat separate also for that reason. And it indeed contains just standalone widgets which can potentially be reused in other applications as well.